spec/controllers/passengers_controller_spec.rb

```diff
@@ -0,0 +1,102 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe PassengersController do
+
+  let!(:admin) { create :user, :admin }
+  before :each do
+    session[:user_id] = admin.id
+  end
+
+  describe 'POST set_status' do
+    context 'with broken mailer' do
+      context 'changing active to archived' do
+        it 'saves the passenger and displays a warning message' do
+          allow(PassengerMailer).to receive(:notify_archived).and_raise Net::SMTPFatalError
+
+          @passenger = create :passenger, :permanent, active_status: 'active'
+          post :set_status, params: { id: @passenger.id, status: 'archived' }
+          @passenger.reload
+
+          expect(@passenger.active_status).to eq 'archived'
+          expect(response).to redirect_to(passengers_path)
+          expect(flash[:warning]).to match(/the passenger was unable to be notified of their status change via email/)
```

Comment:
I'm always a little wary of super-log regexes for UI strings. Pare it down a bit to the essence of the warning?